### PR TITLE
Add Kafka Json encoder

### PR DESCRIPTION
**Pull request by yangy0000**
Cherry-pick of https://github.com/prestosql/presto/commit/85204c1bbbe3df2dacd25d8db4b9178de555c052#diff-822c322b24ab891f0ad5abf372f197351788e75be3d6b3cb11af01d7d9d6621e

```
== RELEASE NOTES ==

General Changes
* ...
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/etc/catalog/kafka.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+connector.name=kafka
+kafka.table-names=tpch.kafka_availability_dca1d,kafka_availability_dca1d,kafka-availability-dca1d,hp.event.user.driver_app,test_topic
+kafka.nodes=localhost:9092
+
+kafka.hide-internal-columns=false
+kafka.table-description-dir=/Users/yayang/Code/presto/kafka
```
Comment (yupeng9) [inline presto-main/etc/catalog/kafka.properties:6]: remove this

Comment (yangy0000) [reply presto-main/etc/catalog/kafka.properties]: done

File: presto-kafka/pom.xml [hunk 2]
```diff
@@ -108,7 +108,6 @@
             <artifactId>presto-common</artifactId>
             <scope>provided</scope>
         </dependency>
-
```
Comment (beinan) [inline presto-kafka/pom.xml:111]: keep the blank line

File: presto-kafka/src/test/java/com/facebook/presto/kafka/TestKafkaIntegrationSmokeTest.java [hunk 3]
```diff
@@ -49,25 +49,25 @@ public TestKafkaIntegrationSmokeTest(EmbeddedKafka embeddedKafka)
...
-    @Test(dataProvider = "testRoundTripAllFormatsDataProvider")
+    @Test(dataProvider = "roundTripAllFormatsDataProvider")
     public void testRoundTripAllFormats(RoundTripTestCase testCase)
     {
         assertUpdate("INSERT into write_test." + testCase.getTableName() +
                 " (" + testCase.getFieldNames() + ")" +
                 " VALUES " + testCase.getRowValues(), testCase.getNumRows());
         assertQuery("SELECT " + testCase.getFieldNames() + " FROM write_test." + testCase.getTableName() +
-                " WHERE f_bigint > 1",
+                        " WHERE f_bigint > 1",
```
Comment (beinan) [inline presto-kafka/src/test/java/com/facebook/presto/kafka/TestKafkaIntegrationSmokeTest.java:59]: remove the extra tab

**Review by beinan — COMMENTED** [review]
Looks good to me except a couple format issues

**Review by zhenxiao — APPROVED** [review]
LGTM @yangy0000 could you please squash the 3 commits into 1?

**Pull request merged**